docs/articles/migration/350-400.md

```diff
@@ -0,0 +1,278 @@
+# 3.5.0-4.0.0 Nebular Migration
+
+## Migration Purposes
+Nebular 4.0 is a significant improvement and rework of the previous versions. We believe these changes bring Nebular to the new level of quality and feature-richness with:
+
+- Updated look&feel with Eva Design System;
+- New convenient and flexible Theming system;
+- Custom CSS properties support in theming;
+- Unified component interfaces (`@Input`'s and `@Output`'s);
+- New Dark theme;
+- Support of SVG Eva Icons.
+
+To achieve all of this and allow Nebular to grow faster and better, we had to refactor a lot of internal implementations, as well as *public APIs*.
+
+We hope that next Nebular versions won't receive such significant upgrades and the amount of breaking changes will be kept as low as possible.
+<hr>
+
+## Nebular 3 LTS
+
+Starting with Nebular 3 we introduce support policy schedule. 
+All major versions will receive 6 months of long-time support (LTS):
+
+- critical bug fixes
+- security fixes
+- Angular major version update
+
+This means that Nebular 3 will be maintained for the next 6 months until Angular 9 release.
+We anticipate 3.6.0 version supporting Angular 8 in the following weeks.
+
+<hr>
+
+## Migration Steps
+
+The following migration steps are required to update:
+
+- Update Angular to version 8 *required*
+- Update Nebular to version 4 *required*
+- Fix imports and component properties *required*
+- Migrate Theme *depends*
+- Migrate to `eva-icons`  *optional*
+
+<hr>
+
+###### 1. Update Angular to version 8
+Nebular 4.0 is based on Angular version 8, thus the very first step is to update the last.
+
+Since Angular has an automatic migration `ng update`  in place, the update will look as follows:
+```sh
+ng update @angular/cli @angular/core
+```
+
+Most of the changes will be done automatically.
+If you have any issues or your setup is more complex - refer to the https://update.angular.io/, select your Angular version and follow the steps
+
+<hr>
+
+###### 2. Update Nebular to version 4
+
+```sh
+ng update @nebular/theme
+```
+
+Additionally, if you use any other Nebular packages, you can add them like this:
+
+```sh
+ng update @nebular/theme @nebular/auth @nebular/security
+```
+
+Since Nebular requires `@angular/cdk`, install the latest version:
```

Comment:
3.5.0 already depends on cdk. Shouldn't it be rather listen among other Angular packages to update on line 49.

Reply:
the order matters, it won't let you I believe